### DISCUSSION TITLE
Front-End - XML, CSS, O Browser e outros Sistemas Operacionais

### DISCUSSION THREAD

**Discussion by felquis** [frontendbr/forum#523] [Discussão]
Olá pessoal, hoje tive uma longa conversa com o @derekstavis sobre como o CSS altera a UI do GNOME, e como ele usa XML para criar uma estrutura de árvore para o conteúdo e o CSS seleciona estes tagNames do XML e aplica propriedades de estilo.

Isso é exatamente a mesma coisa que vejo acontecer no HTML e CSS, e vejo que já buscamos abstrair estes dois com várias tecnolocias como frameworks e libraries das mais variadas formas possíveis, como sou relativamente novo no mundo do software development tem 6 anos que eu fiquei fazendo layouts para web resumidamente, tem várias coisas que não conheço sobre como as interfaces funcionam no geral e sinceramente não só na parte do software, mas desde a concepção de uma tela desde o atomo, até as formas mais baixo nível de fazer uma tela funcionar, até chega ao mais alto nível que é caraca, escrever em um browser.. :bomb: MIND BLOWING :boom:  não só no browser porém em outros sistemas operacionais, e é justamente este tópico que eu gostaria de puxar para ver se a gente consegue elevar nossa barra técnica a algum outro lugar.

Como transformar o grupo que somos hoje em pessoas especialistas não só em layouts para web, mas layouts para as mais variadas plataformas, iOS, Android, Browsers Desktop e Mobile, Windows, MACOS, e também outros tipos sistemas que possuem telas que eu possa imaginar no momento.

Então, pessoas mais esperientes, respondam a pergunta: quais são os componentes necessários para renderizar uma interface em um dado sistema operacional? Também é importante saber como o sistema operacional usa isso.. Por exemplo, qual é o agnostico do Webkit ou Gecko fora de um browser e sim no Windows por exemplo? Existe este tipo de relação? Se sim, aponta a direção para as pessoas que são mais leigas por favor, ou colabore na pesquisa.

Pode ser inclusive listinha de perguntas para se fazer ao DuckDuckGo para aprender mais sobre este tipo de assunto! 

:smile::gem:

**Comment by bernardodiasc** [comment]
To muito curioso pra aprender frontend pra outros tipos de telas e sistemas, seguindo essa issue!

Atualmente to estagnado fazendo interface pra browser, minha sorte na ultima década foi que tem browser em tudo que é device e tanto tempo usando html e css deixa tudo bem facil fazer. Tive um bocado de expericencia usando Flash, fiz diversas animações, sites inteiros com tudo que um site em React tem hoje em dia (ok, nem tudo 😝), muito ActionScript, etc. Usávamos muito pra gerar apresentações de CDRom pra empresas compilando `.swf` pra `.exe` e includingo no auto init do CD. Isso pra não voltar mais ainda no tempo dos Java Applets e DHTML para fazer animações.

E tambem tive bastante experiencia com [XSLT](https://github.com/bernardodiasc/xslt), pra transformar XML em HTML. O interessante do XSLT é que se aplica aos dados e o browser renderiza aquilo na hora, basicamente um "transpiler" nativo dos browsers que existe desde 99 e até a data de hoje ainda funciona (se não me engano o Chrome tem planos de dropar suporte ((sera que dropam o XML/HTML junto? huahua "WebAssembly" 👻)), já no W3C o contrário, XML-family ta firme e forte, mas os ventos sopram na direção do 💰, sei lá...). To ligado que as técnicas evoluíram muito para renderizar DOM, mas ainda assim essa técnica old school é muito maneira.

Espero ter oportunidades de brincar com React Native em breve e quem sabe Cordova ou alguma outra coisa do tipo, sendo HTML, CSS e JS fica facil, mas linguagens mais baixas ja fica fora da minha ossada. Ja tive experiencia de construir uma maquina com circuitos, passando a canetinha na placa pra desenhar or circuitos, mergulhando ela no ácido e depois soldando chips e resistors e depois escrevendo código simplão em Assembly pra então renderizar em display de 8 segmentos usando swich-buttons para input do usuário. Complicado, deu certo mas demorou semanas, foi bem difícil, mas interessante. A curiosidade de como são renderizados em sistemas operacionais sempre tive, conheci uns programadores de C++ que manjavam de temas para Linux e essas coisas, mas não tive oportunidade de aprender.
